### ISSUE TITLE
[BUG] wrapper for transformers.TextClassificationPipeline quietly strips away the text_pair argument when invoked on a pair of texts

### ISSUE THREAD

**Issue by machinehead** [mlflow/mlflow#9171]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

2.5.0

### System information

will be reproducible on any system

### Describe the problem

I'm trying to save a transformers text-classification pipeline via mlflow.transformers.log_model.

Calling "predict" on the resulting model with a pair of texts (e.g. with {"text": ..., "text_pair": ...}) will always lose the 2nd text due to this line in `mlflow.transformers._TransformersWrapper._parse_raw_pipeline_input`:

> return [list(entry.values())[0] for entry in data]

This line generally looks unsafe, as it's potentially relying on the order of insertion of the keys in the "entry" dict passed as input to `mlflow.transformers._TransformersWrapper.predict`.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
WILL PROVIDE A BIT LATER
```

### Stack trace

```
There's no stack trace, the error is silent and results in incorrect predictions.
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by BenWilson2** [comment]
Hi @machinehead thank you for bringing up this oversight. I'll make sure to get this fixed prior to the release of Mlflow 2.6 :)

**Comment by BenWilson2** [comment]
HI @machinehead the fix has been merged and will be included in the next release!
